### PR TITLE
bitmanipulation: link unit tests to comp_req requirements

### PR DESCRIPTION
**Pull request by 4og**
Link all `bit_manipulation` and `bitmask_operators` unit tests to the `comp_req__bitmanipulation__*` requirements they verify, using `FullyVerifies`/`PartiallyVerifies`, `TestType`, `DerivationTechnique`, and `Description` metadata per the `test-to-requirement-linking` skill.

Also sharpens the skill's own guidance:
- Tightens the `equivalence-classes` vs `boundary-values` distinction with a concrete decision rule.
- Adds a `TestType` breakdown table (previously only `DerivationTechnique` had one).
- Notes that a passing `bazel test` does not confirm a linked requirement ID actually resolves; only `bazel run //:docs` validates that.

### CODE REVIEW

File: score/bitmanipulation/bit_manipulation_test.cpp [hunk 1]
```diff
@@ -151,6 +234,12 @@ TEST(SetBit, WithUInt64)
 
 TEST(SetBit, OverflowWithUInt8)
 {
+    RecordProperty("PartiallyVerifies", "comp_req__bitmanipulation__bounds_safety");
```
Comment (fbaeuerle) [inline score/bitmanipulation/bit_manipulation_test.cpp:254]: From the requirement:
>   The bit manipulation functions shall validate input parameters against bounds and, on out-of-bounds access, shall leave the target value unmodified and return false.

it is nowhere verified (in all tests linked to that requirement) that the target value remains unmodified

File: score/bitmanipulation/bitmask_operators_test.cpp [hunk 2]
```diff
@@ -41,15 +42,13 @@ namespace
 {
 using UnderlyingType = std::underlying_type_t<MyBitmask>;
 
-TEST(MyBitmask, UnderlyingValuesMatchExpectations)
```
Comment (fbaeuerle) [inline score/bitmanipulation/bitmask_operators_test.cpp:44]: Is this test nonsense or is there no requirement for it?

**Review by fbaeuerle — CHANGES_REQUESTED** [review]
linkage looks good, but there's a a testing gap (which now becomes obvious when reading the linked requirement)

File: score/bitmanipulation/bitmask_operators_test.cpp (hunk 2 above)
Comment (4og) [reply score/bitmanipulation/bitmask_operators_test.cpp:44]: It is nonsense. The enum `MyBitmask` is defined on line 24 in this file

File: score/bitmanipulation/bit_manipulation_test.cpp (hunk 1 above)
Comment (4og) [reply score/bitmanipulation/bit_manipulation_test.cpp:254]: Fixed

File: score/bitmanipulation/bit_manipulation_test.cpp [hunk 3]
```diff
@@ -28,41 +28,78 @@ using ::testing::Eq;
 
 TEST(HalfByte, CanBeConstructedFromUInt8)
 {
+    RecordProperty("PartiallyVerifies", "comp_req__bitmanipulation__byte_operations");
```
Comment (fbaeuerle) [inline score/bitmanipulation/bit_manipulation_test.cpp:31]: found a mismatch between implementation and requirement:
>   The bit manipulation component shall provide an API for extracting and setting bytes and half-bytes for any integral type up to 64 bits, returning boolean success status.

it doesn't quite match the implementation, we have the byte and halfbyte types. we also have `GetByte`, which does not return bool (and i wonder how it could possibly return a failure). we do however, do not have a `SetByte` counterpart. and even on the `SetByte`counterpart i see little value in reporting success/failure as a boolean (all errors i can think of can be caught during compilation).

Comment (4og) [reply score/bitmanipulation/bit_manipulation_test.cpp:31]: I addressed this by splitting the requirement. The bool status is removed, and the  "setting bytes" is also removed.

Comment (fbaeuerle) [reply score/bitmanipulation/bit_manipulation_test.cpp:31]: https://github.com/eclipse-score/baselibs/blob/main/score/bitmanipulation/docs/requirements/chklst_req_inspection.rst?plain=1#L137

does this now need an update? the linked finding (https://github.com/eclipse-score/score/issues/2698) was already closed (i see back then you added those requirements)

**Review by fbaeuerle — APPROVED** [review]

File: score/bitmanipulation/bit_manipulation_test.cpp (hunk 3 above)
Comment (4og) [reply score/bitmanipulation/bit_manipulation_test.cpp:31]: Eventually, we will need to do the re-inspection. I dropped the `:tags: inspected` to indicate that new requirement inspection is needed
